### ISSUE TITLE
False positive for `IfUnlessModifier` with pattern matching

### ISSUE THREAD

**Issue by mehanoid** [rubocop/rubocop#11723]
If an `if` condition contains a pattern matching with a variable assignment, rubocop may erroneously suggest changing this code to use an `if` modifier, but the changed code does not work.

--------

## Expected behavior

No offences, no autocorrection

## Actual behavior

```
For /home/meh/Projects/sandbox: Default configuration from /home/meh/.rbenv/versions/3.2.1/lib/ruby/gems/3.2.0/gems/rubocop-1.48.1/config/default.yml
Use parallel by default.
Skipping parallel inspection: only a single file needs inspection
Inspecting 1 file
Scanning /home/meh/Projects/sandbox/rubocop_pattern_match.rb
C

Offenses:

rubocop_pattern_match.rb:3:1: C: [Correctable] Style/IfUnlessModifier: Favor modifier if usage when having a single-line body. Another good alternative is the usage of control flow &&/||.
if [1] in [x]
^^

1 file inspected, 1 offense detected, 1 offense autocorrectable
Finished in 0.1424515729995619 seconds
```

Code after autocorrection:
```ruby
# frozen_string_literal: true

puts x if [1] in [x]
```
This code does not work because the variables declared to the right of if are not visible to the code to the left of if. As a result, we get an error:
```
undefined local variable or method `x' for main:Object (NameError)

puts x if [1] in [x]
     ^
```

## Steps to reproduce the problem

```ruby
# frozen_string_literal: true

if [1] in [x]
  puts x
end
```

## RuboCop version

```
$ rubocop -V
1.48.1 (using Parser 3.2.1.0, rubocop-ast 1.27.0, running on ruby 3.2.1) [x86_64-linux]
```
